### ISSUE TITLE
Not enough models to train Augustus.

### ISSUE THREAD

**Issue by kimnegrette3** [nextgenusfs/funannotate#418]
**Are you using the latest release?**
I am using funannotate v1.7.4 downloaded via conda 4.7.11

**Describe the bug**
**Genome has not enough gene models to train Augustus. It requires 200, and has only 37.** 
My genome is 45M in size, from a microalgae, genus Scenedesmus. I only have a genome asembly and I am trying to predict and annotate it. I changed some options like --organism and --busco_db, as suggested in the tutorial for no-fungi organisms, also gave the option busco_seed_species with chlamydomonas which is the closest organism in the available species list to train Augustus. 

**What command did you issue?**
funannotate predict -i Sdimorphus_sorted_masked.fasta  -o Sdim1 --species "Scenedesmus dimorphus" --busco_seed_species chlamydomonas --organism other --busco_db eukaryota

**Logfiles**
_funannotate-predict.log_
[07:10 PM]: OS: linux2, 56 cores, ~ 528 GB RAM. Python: 2.7.15
[07:10 PM]: Running funannotate v1.7.4
[07:10 PM]: GeneMark not found and $GENEMARK_PATH environmental variable missing. Will skip GeneMark ab-initio predicti$[07:10 PM]: Parsed training data, run ab-initio gene predictors as follows:
[07:10 PM]: CodingQuarry will be skipped --> --rna_bam required for training
[07:10 PM]: Loading genome assembly and parsing soft-masked repetitive sequences
  ^[[4mProgram      Training-Method^[[0m
  augustus     busco
  glimmerhmm   busco
  snap         busco
[07:10 PM]: Genome loaded: 1,446 scaffolds; 45,689,722 bp; 4.22% repeats masked
[07:10 PM]: Mapping 549,368 proteins to genome using diamond and exonerate
[07:29 PM]: Found 2,132,765 preliminary alignments --> aligning with exonerate
     Progress: 0.03% ^M     Progress: 0.03% ^M     Progress: 0.03% ^M     Progress: 0.03% ^M     Progress: 0.03% ^M    $[09:13 PM]: Running BUSCO to find conserved gene models for training ab-initio predictors
[09:21 PM]: 35 valid BUSCO predictions found, now formatting for EVM
[09:22 PM]: Running EVM commands with 1 CPUs
     Progress: 0.00% ^M     Progress: 0.00% ^M     Progress: 0.00% ^M     Progress: 0.00% ^M     Progress: 0.00% ^M    $[09:29 PM]: 37 total gene models from EVM, now validating with BUSCO HMM search
[09:30 PM]: 37 BUSCO predictions validated
[09:30 PM]: Not enough gene models 37 to train Augustus (200 required), exiting


_And the busco.log_

INFO    ****************** Start a BUSCO 2.0 analysis, current time: 05/02/2020 21:13:35 ******************
INFO    The lineage dataset is: eukaryota_odb9 (eukaryota)
INFO    Mode is: genome
INFO    Maximum number of regions limited to: 3
INFO    To reproduce this run: python /hpcfs/home/je.gutierrez11/.conda/envs/funannotate/lib/python2.7/site-packages/funannotate/aux_scripts/funannotate-BUSCO2.py -i /hpcfs/home/je.gutierrez11/funannotate/Sdim1$INFO    Check dependencies...
INFO    Check input file...
INFO    Temp directory is ./tmp/

INFO    ****** Phase 1 of 2, initial predictions ******
INFO    ****** Step 1/3, current time: 05/02/2020 21:13:38 ******
INFO    Create blast database...
INFO    [makeblastdb]   Building a new DB, current time: 05/02/2020 21:13:39
INFO    [makeblastdb]   New DB name:   /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco/tmp/scenedesmus_dimorphus_4180716481
INFO    [makeblastdb]   New DB title:  /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/genome.softmasked.fa
INFO    [makeblastdb]   Sequence type: Nucleotide
INFO    [makeblastdb]   Keep Linkouts: T
INFO    [makeblastdb]   Keep MBits: T
INFO    [makeblastdb]   Maximum file size: 1000000000B
INFO    [makeblastdb]   Adding sequences from FASTA; added 1446 sequences in 0.590258 seconds.
INFO    Running tblastn, writing output to /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco/run_scenedesmus_dimorphus/blast_output/tblastn_scenedesmus_dimorphus.tsv...
INFO    ****** Step 2/3, current time: 05/02/2020 21:13:57 ******
INFO    Getting coordinates for candidate regions...
INFO    Pre-Augustus scaffold extraction...
INFO    Running Augustus prediction using chlamydomonas as species:
INFO    [augustus] Please find all logs related to Augustus here: /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco/run_scenedesmus_dimorphus/augustus_output/augustus.log
INFO    05/02/2020 21:13:58 =>  0% of predictions performed (271 to be done)
INFO    05/02/2020 21:14:33 =>  10% of predictions performed (30/271 candidate regions)
INFO    05/02/2020 21:15:03 =>  20% of predictions performed (57/271 candidate regions)
INFO    05/02/2020 21:15:36 =>  30% of predictions performed (85/271 candidate regions)
INFO    05/02/2020 21:16:08 =>  40% of predictions performed (112/271 candidate regions)
INFO    05/02/2020 21:16:38 =>  50% of predictions performed (139/271 candidate regions)
INFO    05/02/2020 21:17:10 =>  60% of predictions performed (166/271 candidate regions)
INFO    05/02/2020 21:17:45 =>  70% of predictions performed (193/271 candidate regions)
INFO    05/02/2020 21:18:16 =>  80% of predictions performed (220/271 candidate regions)
INFO    05/02/2020 21:18:48 =>  90% of predictions performed (247/271 candidate regions)
INFO    05/02/2020 21:19:13 =>  100% of predictions performed
INFO    Extracting predicted proteins...
INFO    ****** Step 3/3, current time: 05/02/2020 21:19:20 ******
INFO    Running HMMER to confirm orthology of predicted proteins:
INFO    05/02/2020 21:19:20 =>  0% of predictions performed (145 to be done)
INFO    05/02/2020 21:19:21 =>  10% of predictions performed (16/145 candidate proteins)
INFO    05/02/2020 21:19:21 =>  20% of predictions performed (32/145 candidate proteins)
INFO    05/02/2020 21:19:22 =>  30% of predictions performed (45/145 candidate proteins)
INFO    05/02/2020 21:19:22 =>  40% of predictions performed (61/145 candidate proteins)
INFO    05/02/2020 21:19:22 =>  50% of predictions performed (74/145 candidate proteins)
INFO    05/02/2020 21:19:23 =>  60% of predictions performed (89/145 candidate proteins)
INFO    05/02/2020 21:19:23 =>  70% of predictions performed (104/145 candidate proteins)                                                                                                                          INFO    05/02/2020 21:19:23 =>  80% of predictions performed (118/145 candidate proteins)
INFO    05/02/2020 21:19:24 =>  90% of predictions performed (133/145 candidate proteins)
INFO    05/02/2020 21:19:24 =>  100% of predictions performed
INFO    Results:
INFO    C:15.9%[S:11.6%,D:4.3%],F:2.3%,M:81.8%,n:303                                                                                                                                                               INFO    48 Complete BUSCOs (C)                                                                                                                                                                                     INFO    35 Complete and single-copy BUSCOs (S)
INFO    13 Complete and duplicated BUSCOs (D)                                                                                                                                                                      INFO    7 Fragmented BUSCOs (F)                                                                                                                                                                                    INFO    248 Missing BUSCOs (M)                                                                                                                                                                                     INFO    303 Total BUSCO groups searched                                                                                                                                                                                                                                                                                                                                                                                               INFO    ****** Phase 2 of 2, predictions using species specific training ******
INFO    ****** Step 1/3, current time: 05/02/2020 21:19:24 ******
INFO    Extracting missing and fragmented buscos from the ancestral_variants file...                                                                                                                               INFO    Running tblastn, writing output to /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco/run_scenedesmus_dimorphus/blast_output/tblastn_scenedesmus_dimorphus_missing_and_frag_rerun.tsv...
INFO    Getting coordinates for candidate regions...                                                                                                                                                               INFO    ****** Step 2/3, current time: 05/02/2020 21:21:39 ******
INFO    Training Augustus using Single-Copy Complete BUSCOs:
INFO    05/02/2020 21:21:39 =>  Converting predicted genes to short genbank files...
INFO    05/02/2020 21:21:42 =>  All files converted to short genbank files, now running the training scripts...                                                                                                    INFO    Pre-Augustus scaffold extraction...
INFO    Re-running Augustus with the new metaparameters, number of target BUSCOs: 255
INFO    05/02/2020 21:21:44 =>  0% of predictions performed (216 to be done)
INFO    05/02/2020 21:21:45 =>  10% of predictions performed (25/216 candidate regions)
INFO    05/02/2020 21:21:45 =>  20% of predictions performed (46/216 candidate regions)
INFO    05/02/2020 21:21:46 =>  30% of predictions performed (67/216 candidate regions)
INFO    05/02/2020 21:21:46 =>  40% of predictions performed (89/216 candidate regions)
INFO    05/02/2020 21:21:47 =>  50% of predictions performed (112/216 candidate regions)
INFO    05/02/2020 21:21:47 =>  60% of predictions performed (132/216 candidate regions)
INFO    05/02/2020 21:21:48 =>  70% of predictions performed (154/216 candidate regions)
INFO    05/02/2020 21:21:48 =>  80% of predictions performed (175/216 candidate regions)
INFO    05/02/2020 21:21:49 =>  90% of predictions performed (197/216 candidate regions)
INFO    05/02/2020 21:21:49 =>  100% of predictions performed
INFO    Extracting predicted proteins...
INFO    ****** Step 3/3, current time: 05/02/2020 21:21:53 ******
INFO    Running HMMER to confirm orthology of predicted proteins:
INFO    05/02/2020 21:21:53 =>  0% of predictions performed (0 to be done)
INFO    05/02/2020 21:21:53 =>  100% of predictions performed
INFO    Results:
INFO    C:15.9%[S:11.6%,D:4.3%],F:2.3%,M:81.8%,n:303                                                                                                                                                               INFO    48 Complete BUSCOs (C)                                                                                                                                                                                     INFO    35 Complete and single-copy BUSCOs (S)
INFO    13 Complete and duplicated BUSCOs (D)                                                                                                                                                                      INFO    7 Fragmented BUSCOs (F)                                                                                                                                                                                    INFO    248 Missing BUSCOs (M)                                                                                                                                                                                     INFO    303 Total BUSCO groups searched                                                                                                                                                                                                                                                                                                                                                                                               INFO    BUSCO analysis done. Total running time: 499.031779051 seconds                                                                                                                                             INFO    Results written in /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco/run_scenedesmus_dimorphus/


INFO    ****************** Start a BUSCO 2.0 analysis, current time: 05/02/2020 21:29:58 ******************                                                                                                        INFO    The lineage dataset is: eukaryota_odb9 (eukaryota)
INFO    Mode is: proteins
INFO    To reproduce this run: python /hpcfs/home/je.gutierrez11/.conda/envs/funannotate/lib/python2.7/site-packages/funannotate/aux_scripts/funannotate-BUSCO2.py -i /hpcfs/home/je.gutierrez11/funannotate/Sdim1$INFO    Check dependencies...                                                                                                                                                                                      INFO    Check input file...                                                                                                                                                                                        INFO    Temp directory is ./tmp/                                                                                                                                                                                   INFO    Running HMMER on the proteins:                                                                                                                                                                             INFO    05/02/2020 21:29:58 =>  0% of predictions performed (303 to be done)
INFO    05/02/2020 21:29:59 =>  10% of predictions performed (35/303 candidate proteins)
INFO    05/02/2020 21:30:00 =>  20% of predictions performed (64/303 candidate proteins)
INFO    05/02/2020 21:30:01 =>  30% of predictions performed (95/303 candidate proteins)                                                                                                                           INFO    05/02/2020 21:30:01 =>  40% of predictions performed (126/303 candidate proteins)
INFO    05/02/2020 21:30:02 =>  50% of predictions performed (156/303 candidate proteins)
INFO    05/02/2020 21:30:03 =>  60% of predictions performed (186/303 candidate proteins)
INFO    05/02/2020 21:30:03 =>  70% of predictions performed (216/303 candidate proteins)
INFO    05/02/2020 21:30:04 =>  80% of predictions performed (246/303 candidate proteins)
INFO    05/02/2020 21:30:05 =>  90% of predictions performed (276/303 candidate proteins)
INFO    05/02/2020 21:30:05 =>  100% of predictions performed                                                                                                                                                      INFO    Results:                                                                                                                                                                                                   INFO    C:11.6%[S:11.6%,D:0.0%],F:0.0%,M:88.4%,n:303                                                                                                                                                               INFO    35 Complete BUSCOs (C)                                                                                                                                                                                     INFO    35 Complete and single-copy BUSCOs (S)                                                                                                                                                                     INFO    0 Complete and duplicated BUSCOs (D)                                                                                                                                                                       INFO    0 Fragmented BUSCOs (F)                                                                                                                                                                                    INFO    268 Missing BUSCOs (M)                                                                                                                                                                                     INFO    303 Total BUSCO groups searched                                                                                                                                                                                                                                                                                                                                                                                               INFO    BUSCO analysis done. Total running time: 7.40513420105 seconds                                                                                                                                             INFO    Results written in /hpcfs/home/je.gutierrez11/funannotate/Sdim1/predict_misc/busco_proteins/run_scenedesmus_dimorphus/


I would really appreciate if you can give some advice on what should I do to improve this result. 
Best regards,

Kimberly.

**Comment by nextgenusfs** [comment]
It seems that BUSCO is not finding very many gene models. There is an option to override the minimum models required for training, ie you can adjust `--min_training_models`, however it is still expecting to find at least 100 for training so I think it may still die. 

Sometimes BUSCO exists early/silently due to tblastn multithreading issue, I thought we had this fixed and I pinned an old version in the Conda recipe, so I don't think that's the issue. 

But this might be indicative of an incomplete de novo assembly or that the generalized "eukaryote" busco models don't well with your organism. Are any of the sub groups more appropriate for busco models?  You could also try a different busco_seed_species and see if that will improve results. 

```
$ funannotate database --show-buscos
-----------------------------
BUSCO DB tree: (# of models)
-----------------------------
eukaryota (303)
	metazoa (978)
		nematoda (982)
		arthropoda (1066)
			insecta (1658)
			endopterygota (2442)
			hymenoptera (4415)
			diptera (2799)
		vertebrata (2586)
			actinopterygii (4584)
			tetrapoda (3950)
			aves (4915)
			mammalia (4104)
		euarchontoglires (6192)
			laurasiatheria (6253)
	fungi (290)
		dikarya (1312)
			ascomycota (1315)
				pezizomycotina (3156)
					eurotiomycetes (4046)
					sordariomycetes (3725)
					saccharomycetes (1759)
						saccharomycetales (1711)
			basidiomycota (1335)
		microsporidia (518)
	embryophyta (1440)
	protists (215)
		alveolata_stramenophiles (234)
```

**Comment by hyphaltip** [comment]
So try with embryophyta_odb9 first in case there are still enough shared orthologs in the algae.
 
You may try the busco10 set as more orthologroups are present. I was able to run it just with simple download of the busco10 datasets and the procedures still work (either put the full path to the odb10 downloaded folder or you can put it into the $FUNANNOTATE_DB folder
You can try viridiplantae Odb10 or chlorophyta_obd10
https://busco.ezlab.org/frames/plants.htm

You can also set  --min_training_models 100 in funannotate to proceed with fewer models. But augustus will fail if it cannot split into train / test datasets with enough models so this may not really solve it for you.

I have also gone about training augustus outside of funannotate with fragmented genome but usually I have some RNAseq to support - do you have that as well from this or a related species?

**Comment by kimnegrette3** [comment]
Hi! thank you both for your prompt reply!
I don´t have RNAseq for this or any closely related species, so I will first try with the  embryophyta_odb9 and chlorophyta_obd10 datasets and see if that works for me. Thanks again and I´ll let you know if this solves the problem.

**Comment by kimnegrette3** [comment]
Hi! the prediction worked fine with chlorophyta dataset and setting --min_training_models 100. 
Thanks a lot!
